CRM/Activity/Form/Task.php

```diff
@@ -116,10 +116,16 @@ public static function preProcessCommon(&$form, $useTable = FALSE) {
 
       // CRM-12675
       $activityClause = NULL;
-      if (! CRM_Core_Permission::check('access CiviContribute')) {
-        $activityClause = ' (activity_type.component_id IS NULL OR activity_type.component_id <> 2) ';
-      }
 
+      $components = CRM_Core_Component::getNames();
+      foreach ($components as $componentID => $componentName) {
+        if (!CRM_Core_Permission::check("access $componentName")) {
+          $componentClause = " (activity_type.component_id IS NULL OR activity_type.component_id <> {$componentID}) ";
+        }
+      }
+      if (!empty($componentClause)) {
+        $activityClause = implode(' AND ', $componentClause);
+      }
```

Comment:
@colemanw : That was a typo(thanks for catching it!). I 've fixed these issues in #6040